tests/helpers_tests/test_condition.py

```diff
@@ -126,264 +130,192 @@ def test_evaluate__returns_null__when_condition_has_an_invalid_type_property(sel
+        user_versions = ['2.9']
+        for user_version in user_versions:
+            evaluator = condition_helper.CustomAttributeConditionEvaluator(
+                semver_greater_than_2_0_condition_list, {'Android': user_version}, self.mock_client_logger)
+            result = evaluator.evaluate(0)
+            custom_err_msg = "Got {} in result. Failed for user version: {}".format(result, user_version)
+            self.assertEquals(result, True, custom_err_msg)
 
-        self.assertRaises(Exception)
+    def test_evaluate__returns_false__when_user_version_greater_than_target_version(self):
+        user_versions = ['2.0.0', '1.9']
+        for user_version in user_versions:
+            evaluator = condition_helper.CustomAttributeConditionEvaluator(
+                semver_greater_than_2_0_condition_list, {'Android': user_version}, self.mock_client_logger)
+            result = evaluator.evaluate(0)
+            custom_err_msg = "Got {} in result. Failed for user version: {}".format(result, user_version)
+            self.assertEquals(result, False, custom_err_msg)
 
-    def test_evaluate__returns_exception__when_user_provided_version_is_invalid10(self):
+    def test_evaluate__returns_None__when_user_version_is_not_string(self):
+        user_versions = [True, 37]
+        for user_version in user_versions:
+            evaluator = condition_helper.CustomAttributeConditionEvaluator(
+                semver_greater_than_2_0_condition_list, {'Android': user_version}, self.mock_client_logger)
+            result = evaluator.evaluate(0)
+            custom_err_msg = "Got {} in result. Failed for user version: {}".format(result, user_version)
+            self.assertIsNone(result, custom_err_msg)
 
-        condition_helper.CustomAttributeConditionEvaluator(
-            semver_equal_2_0_condition_list, {'Android': True}, self.mock_client_logger
-        )
+    def test_evaluate__returns_None__when_user_version_with_invalid_semantic(self):
+        user_versions = ['3.7.2.2', '+']
+        for user_version in user_versions:
+            evaluator = condition_helper.CustomAttributeConditionEvaluator(
+                semver_greater_than_2_0_condition_list, {'Android': user_version}, self.mock_client_logger)
+            result = evaluator.evaluate(0)
+            custom_err_msg = "Got {} in result. Failed for user version: {}".format(result, user_version)
+            self.assertIsNone(result, custom_err_msg)
 
-        self.assertRaises(Exception)
+    def test_compare_user_version_with_target_version_equal_to_0(self):
+        versions = [
+            ('2.0.1', '2.0.1'),
+            ('2.0.0', '2.0.0'),
+            ('2.9', '2.9.1'),
+            ('2.9.9', '2.9.9'),
+            ('2.9.9-beta', '2.9.9-beta'),
+            ('2.1', '2.1.0'),
+            ('2.1', '2.1.215'),
+            ('2', '2.12'),
+            ('2', '2.785.13')
+        ]
+        for user_version, target_version in versions:
```

Comment:
target_version should come before user_version. Even compare method accepts target_version as first arg. This is mis leading. 
Keep target before user in list, in iteration and passing arg